packages/lab/src/number-input/internal/useFormatControlled.ts

```diff
@@ -0,0 +1,69 @@
+import { isValidElement, useEffect, useRef, useState } from "react";
+
+type UseFormatControlledProps<T> = {
+  controlled?: T;
+  default: T;
+  name: string;
+  state: string;
+  format?: (value: T) => T;
+};
+
+export const useFormatControlled = <S>({
+  controlled,
```

Comment:
this appear like a copy of useControlled and internal to NumberInput, if we go down this route, then suspect we move this to utils?

Could this hook not just call the other hook, just return formatted results ?